### PR TITLE
Replace most prints with log statements

### PR DESCRIPTION
**Pull request by stephenroller**
**Patch description**
- Replace most prints with logs.
- Upgrade many logs to be f-strings
- Add support for colored output if the user runs `pip install coloredlogs`

**Testing steps**
Manual testing. CI.

We need to very carefully review this (long) diff because there might be some `f`'s i missed in my f-strings.

image

### CODE REVIEW

File: parlai/core/torch_classifier_agent.py [hunk 1]
```diff
@@ -343,7 +344,7 @@ def __init__(self, opt: Opt, shared=None):
                     'build_model() and build_criterion() need to return the model or criterion'
                 )
             if init_model:
-                print('Loading existing model parameters from ' + init_model)
+                logging.info('Loading existing model parameters from {init_model}')
```
Comment (stephenroller) [inline parlai/core/torch_classifier_agent.py:347]: missed

File: parlai/core/torch_ranker_agent.py [hunk 2]
```diff
@@ -198,12 +199,14 @@ def __init__(self, opt: Opt, shared=None):
...
-            print(f"Total parameters: {total_params:,d} ({train_params:,d} trainable)")
+            logging.info(
+                f"Total parameters: {total_params:,d} ({train_params:,d} trainable)"
+            )
 
             if self.fp16:
                 self.model = self.model.half()
             if init_model:
-                print('Loading existing model parameters from ' + init_model)
+                logging.info('Loading existing model parameters from ' + init_model)
```
Comment (stephenroller) [inline parlai/core/torch_ranker_agent.py:209]: make f-string

File: parlai/scripts/build_candidates.py [hunk 3]
```diff
@@ -42,8 +43,8 @@ def build_cands(opt):
         num_examples = opt['num_examples']
     log_timer = TimeLogger()
 
-    print('[ starting to build candidates from task.. (ex:' + str(num_examples) + ')]')
-    print('[ saving output to {} ]'.format(outfile))
+    logging.info(f'Starting to build candidates from task.. (ex: {num_examples})')
+    logging.info('Saving output to {outfile}')
```
Comment (stephenroller) [inline parlai/scripts/build_candidates.py:47]: missed this one

File: parlai/core/build_data.py [hunk 4]
```diff
@@ -290,7 +293,7 @@ def unzip(path, fname, deleteZip=True):
     :param bool deleteZip:
         If true, the archive will be deleted after extraction.
     """
-    print('unzipping ' + fname)
+    logging.debug('unzipping {fname}')
```
Comment (EricMichaelSmith) [inline parlai/core/build_data.py:296]: forgot f-string

Comment (stephenroller) [reply parlai/core/build_data.py:296]:
```suggestion
    logging.debug(f'unzipping {fname}')
```

**Pull request merged**